src/main/java/gov/cms/mat/cql_elm_translation/service/HumanReadableService.java

```diff
@@ -98,57 +136,301 @@ HumanReadableMeasureInformationModel buildMeasureInfo(Measure measure) {
+        .denominatorExceptions(
+            HumanReadableUtil.getPopulationDescription(
+                measure, PopulationType.DENOMINATOR_EXCLUSION.name()))
+        .numerator(
+            HumanReadableUtil.getPopulationDescription(measure, PopulationType.NUMERATOR.name()))
+        .numeratorExclusions(
+            HumanReadableUtil.getPopulationDescription(
+                measure, PopulationType.NUMERATOR_EXCLUSION.name()))
+        .denominatorExceptions(
+            HumanReadableUtil.getPopulationDescription(
+                measure, PopulationType.DENOMINATOR_EXCEPTION.name()))
+        .definitions(HumanReadableUtil.buildMeasureDefinitions(measure.getMeasureMetaData()))
+        .references(HumanReadableUtil.buildReferences(measure.getMeasureMetaData()))
         .build();
   }
 
-  private List<Reference> buildReferences(List<Reference> references) {
-    return references.stream()
-        .map(
-            reference ->
-                new Reference()
-                    .toBuilder()
-                        .id(reference.getId())
-                        .referenceText(htmlEscape(reference.getReferenceText()))
-                        .referenceType(reference.getReferenceType())
-                        .build())
-        .collect(Collectors.toList());
-  }
-
-  List<HumanReadablePopulationCriteriaModel> buildPopCriteria(Measure measure) {
+  List<HumanReadablePopulationCriteriaModel> buildPopCriteria(
+      Measure measure, Set<CQLDefinition> allDefinitions) {
     return measure.getGroups().stream()
         .map(
             group ->
                 HumanReadablePopulationCriteriaModel.builder()
                     .id(group.getId())
                     .name(group.getGroupDescription())
-                    .populations(buildPopulations(group))
+                    .populations(
+                        Stream.concat(
+                                buildPopulations(group, allDefinitions).stream(),
+                                buildStratification(group, allDefinitions).stream())
+                            .toList())
                     .build())
         .collect(Collectors.toList());
   }
 
-  private List<HumanReadablePopulationModel> buildPopulations(Group group) {
+  List<HumanReadablePopulationModel> buildPopulations(
+      Group group, Set<CQLDefinition> allDefinitions) {
     return group.getPopulations().stream()
         .map(
             population ->
                 HumanReadablePopulationModel.builder()
                     .name(population.getName().name())
                     .id(population.getId())
                     .display(population.getName().getDisplay())
-                    .logic(population.getDefinition())
+                    .logic(getCQLDefinitionLogic(population.getDefinition(), allDefinitions))
                     .expressionName(population.getDefinition())
+                    .inGroup(StringUtils.isBlank(population.getDefinition()) ? false : true)
```

Comment:
`isBlank` will return true/false without the ternary operator.
```suggestion
                    .inGroup(!StringUtils.isBlank(population.getDefinition()))
```

Reply:
Changed

Reply:
Changed